File: ru-olymp-train-winter-2007/Problems/20070119/problems/binsearch/unit-preliminary/pas/binsearch.pas
unit binsearch;

interface

function getN: longint;

function query(i: longint): boolean;

procedure answer(i: longint);

implementation

var
	c: longint;
	was: boolean;
	fout: text;

function getN: longint;
begin
	if was then begin
    	assign(fout, '31415926.out');
    	rewrite(fout);
   		writeln(fout, -1);
   		writeln(fout, 'getN called twice');
    	close(fout);
    	halt(0);
	end;

	c := 0;
	was := true;
	getn := 1;
end;

function query(i: longint): boolean;
begin
	if i <> 1 then begin
    	assign(fout, '31415926.out');
    	rewrite(fout);
   		writeln(fout, -1);
   		writeln(fout, 'invalid query');
    	close(fout);
    	halt(0);
	end;

	inc(c);
	if c = 2 then
		query := true
	else
		query := false;
end;

procedure answer(i: longint);
begin
  	assign(fout, '31415926.out');
  	rewrite(fout);
	if (i <> 2) then begin
		writeln(fout, -1);
		writeln(fout, 'invalid answer')
	end else                                                             
		writeln(fout, c);
	close(fout);
	halt(0);
end;

begin
	was := false;
end.

File: ru-olymp-train-winter-2007/Problems/20070119/problems/binsearch/unit-preliminary/c/binsearch.c
#include <stdio.h>
#include <stdlib.h>

static int c;
static int was = 0;
static FILE* fout;

int getN() {
	if (was) {
		fout = fopen("31415926.out", "wt");
		fprintf(fout, "-1\ngetN called twice");
		fclose(fout);
		exit(0);
	}

	c = 0;
	was = 1;
	return 1;
}

int query(int i) {
	if (i != 1) {
		fout = fopen("31415926.out", "wt");
		fprintf(fout, "-1\ninvalid query");
		fclose(fout);
		exit(0);
	}
	c++;
	if (c == 2) {
		return 1;
	} else {
		return 0;
	}
}

void answer(int i) {
	fout = fopen("31415926.out", "wt");
	if (i != 2) {
		fprintf(fout, "-1\ninvalid answer");
	} else {
		fprintf(fout, "%d\n", c);
	}
	fclose(fout);
	exit(0);
}
